code/game/machinery/recharger.dm

```diff
@@ -109,78 +86,16 @@ obj/machinery/recharger
 		update_use_power(1)
 		icon_state = icon_state_idle
 	else
-		if(istype(charging, /obj/item/weapon/gun/energy))
-			var/obj/item/weapon/gun/energy/E = charging
-			if(!E.power_supply.fully_charged())
-				icon_state = icon_state_charging
-				E.power_supply.give(active_power_usage*CELLRATE)
-				update_use_power(2)
-			else
-				icon_state = icon_state_charged
-				update_use_power(1)
-			return
-
-		if(istype(charging, /obj/item/weapon/gun/magnetic))
-			var/obj/item/weapon/gun/magnetic/M = charging
-			if(!M.cell.fully_charged())
-				icon_state = icon_state_charging
-				M.cell.give(active_power_usage*CELLRATE)
-				update_use_power(2)
-			else
-				icon_state = icon_state_charged
-				update_use_power(1)
-			return
-
-		if(istype(charging, /obj/item/weapon/melee/baton))
-			var/obj/item/weapon/melee/baton/B = charging
-			if(B.bcell)
-				if(!B.bcell.fully_charged())
-					icon_state = icon_state_charging
-					B.bcell.give(active_power_usage*CELLRATE)
-					update_use_power(2)
-				else
-					icon_state = icon_state_charged
-					update_use_power(1)
-			else
-				icon_state = icon_state_idle
-				update_use_power(1)
-			return
-
-		if(istype(charging, /obj/item/device/laptop))
-			var/obj/item/device/laptop/L = charging
-			if(!L.stored_computer.battery.fully_charged())
-				icon_state = icon_state_charging
-				L.stored_computer.battery.give(active_power_usage*CELLRATE)
-				update_use_power(2)
-			else
-				icon_state = icon_state_charged
-				update_use_power(1)
-			return
-
-		if(istype(charging, /obj/item/device/flashlight))
-			var/obj/item/device/flashlight/F = charging
-			if(F.cell)
-				if(!F.cell.fully_charged())
-					icon_state = icon_state_charging
-					F.cell.give(active_power_usage*CELLRATE)
-					update_use_power(2)
-				else
-					icon_state = icon_state_charged
-					update_use_power(1)
-			else
-				icon_state = icon_state_idle
-				update_use_power(1)
-			return
-
-		if(istype(charging, /obj/item/weapon/cell))
-			var/obj/item/weapon/cell/C = charging
+		var/obj/item/weapon/cell/C = charging.get_cell()
+		if(istype(C))
 			if(!C.fully_charged())
 				icon_state = icon_state_charging
 				C.give(active_power_usage*CELLRATE)
 				update_use_power(2)
 			else
 				icon_state = icon_state_charged
 				update_use_power(1)
+<<<<<<< HEAD
```

Comment:
Uh oh.

Reply:
Not really. I mean, that's how the mirrorbot works. It tries to merge things. There are merge conflicts. If we were doing this as a "Polaris Sync", we'd have to fix this too.

Reply:
Thanks for the clarity